stdlib/public/core/Misc.swift

```diff
@@ -110,3 +112,99 @@ public func _getTypeByMangledNameInContext(
   genericContext: UnsafeRawPointer?,
   genericArguments: UnsafeRawPointer?)
   -> Any.Type?
+
+@_silgen_name("swift_demangle")
+internal func _demangleImpl(
+  mangledName: UnsafePointer<Int8>?,
+  mangledNameLength: UInt,
+  outputBuffer: UnsafeMutablePointer<Int8>?,
+  outputBufferSize: UnsafeMutablePointer<UInt>?,
+  flags: UInt32
+) -> UnsafeMutablePointer<Int8>?
```

Comment:
`swift_demangle` is a C function. It should be declared in the SwiftShims C headers instead of using `silgen_name`.